scope/marshal.go

```diff
@@ -77,7 +77,12 @@ func (self ScopeUnmarshaller) Unmarshal(
 			new_scope, v)
 		if err == nil {
 			if !utils.IsNil(unmarshaller) {
-				env.Set(k, unmarshalled)
+				switch v := unmarshalled.(type) {
```

Comment:
This is not needed if Replay returns nil.

Reply:
Sorry, i removed this. Forgot to update the commit